### ISSUE TITLE
The escape key should close the right click menu on desktop

### ISSUE THREAD

**Issue by justinmc** [flutter/flutter#98163]
When the right click menu is open, pressing the escape key should close it.  I confirmed that this is the native behavior on Mac and Windows.  I assume Linux is the same but I haven't tried.

### Steps to reproduce

 1. Run any Flutter app with a text field on a desktop platform (example given below).
 2. Right click on the text field.  The context menu shows.
 3. Press the escape key.

Expected: The context menu closes.
Actual: Nothing happens and the context menu remains on the screen.

### Solution
We could add a text editing shortcut for this and it should work for now, but if we allow context menus outside of EditableText (as mentioned in https://github.com/flutter/flutter/issues/90563), then the Shortcuts widget won't receive the escape when nothing is focused and it won't always work.

@gspencergoog Is it possible to receive a Shortcut when nothing is focused?

<details>

<summary>Code</summary>

```dart
import 'package:flutter/material.dart';

void main() => runApp(const MyApp());

class MyApp extends StatelessWidget {
  const MyApp({
    Key? key,
  }) : super(key: key);

  @override
  Widget build(BuildContext context) {
    return MaterialApp(
      title: 'Flutter Demo',
      theme: ThemeData(
        primarySwatch: Colors.blue,
      ),
      home: Scaffold(
        appBar: AppBar(
          title: const Text('Flutter Demo'),
        ),
        body: const Center(
          child: TextField(),
        ),
      ),
    );
  }
}
```

</details>

**Comment by Seirdy** [comment]
This could be considered a component of https://github.com/flutter/flutter/issues/94965, since it doesn't look like this is the only type of "de-selection" that the ESC key should handle in Flutter's built-in widgets.

**Comment by justinmc** [comment]
Good point.  I left a comment on that issue.  If that issue doesn't have the same focus problem that this one does, then it will be even easier to fix.

**Comment by markusaksli-nc** [comment]
Reverted in https://github.com/flutter/flutter/pull/98600

**Comment by gspencergoog** [comment]
@justinmc There's no way to get a shortcut when nothing is focused, but you can listen for key events globally through a handler on`HardwareKeyboard`, and if escape is pressed while the context menu is open, close the menu.
